src/notice.ts

```diff
@@ -74,4 +77,24 @@ export function validateNotice(notice: Notice): void {
   if (!semver.validRange(notice.schemaVersion)) {
     throw new Error(`Schema version ${notice.schemaVersion} is not a valid semver range`);
   }
+}
+
+function validNoticeSemver(version: string) {
+  const components = version.split(' ');
```

Comment:
Is there not a semver parse function you can use? String-based manipulation seems dangerous to me.

Reply:
yes, possible: `semver.toComparators()`